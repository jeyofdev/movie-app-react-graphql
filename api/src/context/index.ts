import PersonService from '../services/Person.service';
import CollectionService from '../services/Collection.service';
import MovieService from '../services/Movie.service';
import KeywordService from '../services/Keyword.service';
import GenreService from '../services/Genre.service';
import SearchService from '../services/Search.service';

export interface IContext {
	dataSource: {
		movies: MovieService;
		collections: CollectionService;
		persons: PersonService;
		keywords: KeywordService;
		genres: GenreService;
		search: SearchService;
	};
}

export const context: IContext = {
	dataSource: {
		movies: new MovieService(),
		collections: new CollectionService(),
		persons: new PersonService(),
		keywords: new KeywordService(),
		genres: new GenreService(),
		search: new SearchService(),
	},
};
