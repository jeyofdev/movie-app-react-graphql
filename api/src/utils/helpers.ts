import request from 'supertest';
import { OptionsInput } from '../__generated__/resolvers-types';

export const formatUrlQuery = (
	baseUrl: string,
	apiKey: string,
	pathParams?: string | number,
	queryString?: OptionsInput,
) => {
	let url = `${baseUrl}/`;
	url += pathParams ?? `${pathParams}`;
	url += `?api_key=${apiKey}&`;

	if (queryString) {
		const allQueryStrings = Object.entries(queryString)
			// eslint-disable-next-line @typescript-eslint/no-unused-vars
			.filter(([_, value]) => value)
			.map(value => value.join('='));

		url += allQueryStrings.join('&');
	}
	return url;
};

export const executeRequestForTesting = async (
	url: string,
	query: string,
	variables: unknown,
	statusCode = 200,
) => {
	const response = await request(url)
		.post('/')
		.set('Accept', 'application/json')
		.send({
			query,
			variables,
		})
		.expect(statusCode);

	return response;
};
