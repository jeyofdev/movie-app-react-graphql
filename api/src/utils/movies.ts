import { MoviesResponse } from '../__generated__/resolvers-types';

const setMoviesResult = (movies: MoviesResponse) => ({
	page: movies.page,
	results: movies.results,
	total_pages: movies.total_pages,
	total_results: movies.total_results,
});

export default setMoviesResult;
