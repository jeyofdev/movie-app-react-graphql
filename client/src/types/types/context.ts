import { LanguageEnum } from '@graphql/__generated__/graphql-type';
import { DarkModeEnum } from '../enums';

export type ThemeContextType = {
	themeMode: DarkModeEnum;
	handleThemeMode: () => void;
};

export type TranslationContextType = {
	currentLocale: LanguageEnum;
	handleLocale: (locale: LanguageEnum) => void;
};
