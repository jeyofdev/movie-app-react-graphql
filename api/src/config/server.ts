import { startStandaloneServer } from '@apollo/server/standalone';
import { readFileSync } from 'fs';
import { ApolloServer } from '@apollo/server';
import { IContext, context } from '../context';
import resolvers from '../resolvers';

const getServer = async (port?: number) => {
	const typeDefs = readFileSync('./src/schema/schema.graphql', {
		encoding: 'utf-8',
	});

	const server = new ApolloServer<IContext>({
		typeDefs,
		resolvers,
	});

	const { url } = await startStandaloneServer(server, {
		context: async () => context,
		listen: { port },
	});

	return { server, url };
};

export default getServer;
