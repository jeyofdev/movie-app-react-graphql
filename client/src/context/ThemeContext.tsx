import { createContext, useState } from 'react';
import { DarkModeEnum } from '../types/enums';
import { ThemeContextType } from '../types/types/context';
import { ThemeContextProviderPropsType } from '../types/types/props';

export const ThemeContext = createContext<ThemeContextType>({
	themeMode: DarkModeEnum.DARK,
	handleThemeMode: () => {},
});

const ThemeContextProvider = ({ children }: ThemeContextProviderPropsType) => {
	const [themeMode, setThemeMode] = useState<DarkModeEnum>(DarkModeEnum.DARK);

	const handleThemeMode = () => {
		setThemeMode(
			themeMode === DarkModeEnum.DARK ? DarkModeEnum.LIGHT : DarkModeEnum.DARK,
		);
	};

	return (
		<ThemeContext.Provider value={{ themeMode, handleThemeMode }}>
			{children}
		</ThemeContext.Provider>
	);
};

export default ThemeContextProvider;
