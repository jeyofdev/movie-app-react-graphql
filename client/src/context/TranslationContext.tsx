import { LanguageEnum } from '@graphql/__generated__/graphql-type';
import { createContext, useState } from 'react';
import { TranslationContextType } from '../types/types/context';
import { ThemeContextProviderPropsType } from '../types/types/props';

export const TranslationContext = createContext<TranslationContextType>({
	currentLocale: LanguageEnum.En,
	handleLocale: () => {},
});

const TranslationContextProvider = ({
	children,
}: ThemeContextProviderPropsType) => {
	const [currentLocale, setcurrentLocale] = useState<LanguageEnum>(
		(localStorage.getItem('locale') as LanguageEnum) ?? LanguageEnum.En,
	);

	const handleLocale = (locale: LanguageEnum) => {
		setcurrentLocale(locale);
	};

	return (
		<TranslationContext.Provider value={{ currentLocale, handleLocale }}>
			{children}
		</TranslationContext.Provider>
	);
};

export default TranslationContextProvider;
