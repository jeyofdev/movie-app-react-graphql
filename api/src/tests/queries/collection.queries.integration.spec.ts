import { afterAll, beforeAll, describe, expect, it } from 'vitest';
import { ApolloServer } from '@apollo/server';
import { IContext } from '../../context';
import getServer from '../../config/server';
import 'dotenv/config';
import { executeRequestForTesting } from '../../utils/helpers';
import collectionsQueryOperations from '../../operations/collection.query.operations';

describe('init server', () => {
	let server: ApolloServer<IContext>;
	let url: string;

	beforeAll(async () => {
		if (!process.env.TEST_API_PORT) {
			throw new Error(
				'The environment variable TEST_API_PORT must be specified',
			);
		}
		({ server, url } = await getServer());
	});

	afterAll(async () => {
		await server.stop();
	});

	describe('Queries collections', () => {
		it('Returns the list of movies in a collection', async () => {
			const response = await executeRequestForTesting(
				url,
				collectionsQueryOperations.moviesByCollection,
				{
					collectionId: 10,
					options: {
						language: 'EN',
					},
				},
			);

			expect(response.error).not.toBeUndefined();
			expect(response.body.data.moviesByCollection).toMatchSnapshot();
		});

		it('returns the translations of a collection', async () => {
			const response = await executeRequestForTesting(
				url,
				collectionsQueryOperations.translationsByCollection,
				{
					collectionId: 10,
					options: {
						language: 'EN',
					},
				},
			);

			expect(response.error).not.toBeUndefined();
			expect(response.body.data.translationsByCollection).toMatchSnapshot();
		});
	});
});
