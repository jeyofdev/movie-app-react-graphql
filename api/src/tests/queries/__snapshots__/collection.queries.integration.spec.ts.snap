// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`init server > Queries collections > Returns the list of movies in a collection 1`] = `
{
  "backdrop_path": "/d8duYyyC9J5T825Hg7grmaabfxQ.jpg",
  "id": 10,
  "name": "Star Wars Collection",
  "overview": "An epic space-opera theatrical film series, which depicts the adventures of various characters \\"a long time ago in a galaxy far, far away….\\"",
  "parts": [
    {
      "adult": false,
      "backdrop_path": "/zqkmTXzjkAgXmEWLRsY4UpTWCeo.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 11,
      "original_language": "en",
      "original_title": "Star Wars",
      "overview": "Princess Leia is captured and held hostage by the evil Imperial forces in their effort to take over the galactic Empire. Venturesome Luke Skywalker and dashing captain Han Solo team together with the loveable robot duo R2-D2 and C-3PO to rescue the beautiful princess and restore peace and justice in the Empire.",
      "popularity": 110.076,
      "poster_path": "/6FfCtAuVAW8XJjZ7eWeLibRLWTw.jpg",
      "release_date": "1977-05-25",
      "title": "Star Wars",
      "video": false,
      "vote_average": 8.207,
      "vote_count": 18657,
    },
    {
      "adult": false,
      "backdrop_path": "/aJCtkxLLzkk1pECehVjKHA2lBgw.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1891,
      "original_language": "en",
      "original_title": "The Empire Strikes Back",
      "overview": "The epic saga continues as Luke Skywalker, in hopes of defeating the evil Galactic Empire, learns the ways of the Jedi from aging master Yoda. But Darth Vader is more determined than ever to capture Luke. Meanwhile, rebel leader Princess Leia, cocky Han Solo, Chewbacca, and droids C-3PO and R2-D2 are thrown into various stages of capture, betrayal and despair.",
      "popularity": 33.433,
      "poster_path": "/2l05cFWJacyIsTpsqSgH0wQXe4V.jpg",
      "release_date": "1980-05-20",
      "title": "The Empire Strikes Back",
      "video": false,
      "vote_average": 8.393,
      "vote_count": 15369,
    },
    {
      "adult": false,
      "backdrop_path": "/ziECpBRIyclmBNaFSWlvCfsKbMD.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1892,
      "original_language": "en",
      "original_title": "Return of the Jedi",
      "overview": "Luke Skywalker leads a mission to rescue his friend Han Solo from the clutches of Jabba the Hutt, while the Emperor seeks to destroy the Rebellion once and for all with a second dreaded Death Star.",
      "popularity": 34.289,
      "poster_path": "/xxCnFmRZ83jHTZsBiceG4IBUGoq.jpg",
      "release_date": "1983-05-25",
      "title": "Return of the Jedi",
      "video": false,
      "vote_average": 7.889,
      "vote_count": 14225,
    },
    {
      "adult": false,
      "backdrop_path": "/wDe8LzwuvHYYiuwyNfxdYQq8ti4.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1893,
      "original_language": "en",
      "original_title": "Star Wars: Episode I - The Phantom Menace",
      "overview": "Anakin Skywalker, a young slave strong with the Force, is discovered on Tatooine. Meanwhile, the evil Sith have returned, enacting their plot for revenge against the Jedi.",
      "popularity": 39.095,
      "poster_path": "/6wkfovpn7Eq8dYNKaG5PY3q2oq6.jpg",
      "release_date": "1999-05-19",
      "title": "Star Wars: Episode I - The Phantom Menace",
      "video": false,
      "vote_average": 6.53,
      "vote_count": 13210,
    },
    {
      "adult": false,
      "backdrop_path": "/abwxHfymXGAbbH3lo9PDEJEfvtW.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1894,
      "original_language": "en",
      "original_title": "Star Wars: Episode II - Attack of the Clones",
      "overview": "Following an assassination attempt on Senator Padmé Amidala, Jedi Knights Anakin Skywalker and Obi-Wan Kenobi investigate a mysterious plot that could change the galaxy forever.",
      "popularity": 32.763,
      "poster_path": "/oZNPzxqM2s5DyVWab09NTQScDQt.jpg",
      "release_date": "2002-05-15",
      "title": "Star Wars: Episode II - Attack of the Clones",
      "video": false,
      "vote_average": 6.551,
      "vote_count": 11980,
    },
    {
      "adult": false,
      "backdrop_path": "/5vDuLrjJXFS9PTF7Q1xzobmYKR9.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 1895,
      "original_language": "en",
      "original_title": "Star Wars: Episode III - Revenge of the Sith",
      "overview": "The evil Darth Sidious enacts his final plan for unlimited power -- and the heroic Jedi Anakin Skywalker must choose a side.",
      "popularity": 38.186,
      "poster_path": "/xfSAoBEm9MNBjmlNcDYLvLSMlnq.jpg",
      "release_date": "2005-05-17",
      "title": "Star Wars: Episode III - Revenge of the Sith",
      "video": false,
      "vote_average": 7.4,
      "vote_count": 12428,
    },
    {
      "adult": false,
      "backdrop_path": "/8BTsTfln4jlQrLXUBquXJ0ASQy9.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 140607,
      "original_language": "en",
      "original_title": "Star Wars: The Force Awakens",
      "overview": "Thirty years after defeating the Galactic Empire, Han Solo and his allies face a new threat from the evil Kylo Ren and his army of Stormtroopers.",
      "popularity": 53.705,
      "poster_path": "/wqnLdwVXoBjKibFRR5U3y0aDUhs.jpg",
      "release_date": "2015-12-15",
      "title": "Star Wars: The Force Awakens",
      "video": false,
      "vote_average": 7.3,
      "vote_count": 18023,
    },
    {
      "adult": false,
      "backdrop_path": "/jOzrELAzFxtMx2I4uDGHOotdfsS.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 181812,
      "original_language": "en",
      "original_title": "Star Wars: The Rise of Skywalker",
      "overview": "The surviving Resistance faces the First Order once again as the journey of Rey, Finn and Poe Dameron continues. With the power and knowledge of generations behind them, the final battle begins.",
      "popularity": 52.369,
      "poster_path": "/db32LaOibwEliAmSL2jjDF6oDdj.jpg",
      "release_date": "2019-12-18",
      "title": "Star Wars: The Rise of Skywalker",
      "video": false,
      "vote_average": 6.388,
      "vote_count": 8754,
    },
    {
      "adult": false,
      "backdrop_path": "/5Iw7zQTHVRBOYpA0V6z0yypOPZh.jpg",
      "genre_ids": [
        12,
        28,
        878,
      ],
      "id": 181808,
      "original_language": "en",
      "original_title": "Star Wars: The Last Jedi",
      "overview": "Rey develops her newly discovered abilities with the guidance of Luke Skywalker, who is unsettled by the strength of her powers. Meanwhile, the Resistance prepares to do battle with the First Order.",
      "popularity": 54.169,
      "poster_path": "/kOVEVeg59E0wsnXmF9nrh6OmWII.jpg",
      "release_date": "2017-12-13",
      "title": "Star Wars: The Last Jedi",
      "video": false,
      "vote_average": 6.837,
      "vote_count": 13995,
    },
  ],
}
`;

exports[`init server > Queries collections > returns the translations of a collection 1`] = `
{
  "id": 10,
  "translations": [
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "",
      },
      "english_name": "Arabic",
      "iso_3166_1": "AE",
      "iso_639_1": "ar",
      "name": "العربية",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Междузвездни войни е поредица от шест филма проследяващи живота на Анакин Скайуокър. Открит от джедая Куай-Гон Джин на пустинната планета Татуин Анакин е все още дете, но постепенно се превръща в силния Дарт Вейдър, дясната ръка на Императорът, който почти успява да унищожи джедайте.",
        "title": "Междузвездни войни (поредица)",
      },
      "english_name": "Bulgarian",
      "iso_3166_1": "BG",
      "iso_639_1": "bg",
      "name": "български език",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Col·lecció de pel·lícules de Star Wars.",
        "title": "Star Wars - Col·lecció",
      },
      "english_name": "Catalan",
      "iso_3166_1": "ES",
      "iso_639_1": "ca",
      "name": "Català",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Star Wars (česky původně též Hvězdné války) je americká multimediální franšíza žánru space opera, vytvořená Georgem Lucasem. Její hlavní částí je devítidílná řada celovečerních filmů, ke které se postupně přidala další díla.",
        "title": "Star Wars (kolekce)",
      },
      "english_name": "Czech",
      "iso_3166_1": "CZ",
      "iso_639_1": "cs",
      "name": "Český",
    },
    {
      "data": {
        "homepage": "",
        "overview": "For meget længe siden i en fjern, fjern galakse....",
        "title": "Star Wars (Samling)",
      },
      "english_name": "Danish",
      "iso_3166_1": "DK",
      "iso_639_1": "da",
      "name": "Dansk",
    },
    {
      "data": {
        "homepage": "http://www.starwars-union.de",
        "overview": "\\"Star Wars™\\" ist die größte Science-Fiction-Saga aller Zeiten, die mit ihren bislang neun filmischen Episoden und den unüberschaubaren Verzweigungen alle Dimensionen der Unterhaltungskultur gesprengt hat. Die Abenteuer von Luke Skywalker, Han Solo und der schönen Prinzessin Leia fesseln die Kinozuschauer und die Leser in aller Welt seit fast 30 Jahren.",
        "title": "Star Wars Filmreihe",
      },
      "english_name": "German",
      "iso_3166_1": "DE",
      "iso_639_1": "de",
      "name": "Deutsch",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Μια επική σειρά διαστημικών ταινιών, που απεικονίζει τις περιπέτειες διαφόρων χαρακτήρων «πολύ καιρό πριν, σε έναν γαλαξία πολύ, πολύ μακρινό…».",
        "title": "Star Wars: Το Έπος των Skywalker - Συλλογή",
      },
      "english_name": "Greek",
      "iso_3166_1": "GR",
      "iso_639_1": "el",
      "name": "ελληνικά",
    },
    {
      "data": {
        "homepage": "https://www.starwars.com/films",
        "overview": "An epic space-opera theatrical film series, which depicts the adventures of various characters \\"a long time ago in a galaxy far, far away….\\"",
        "title": "",
      },
      "english_name": "English",
      "iso_3166_1": "US",
      "iso_639_1": "en",
      "name": "English",
    },
    {
      "data": {
        "homepage": "",
        "overview": "La colección de las 9 películas del director , escritor y productor George Lucas. Con Luke Skywalker, Princesa Leia, Darth Vader,C3PO,R2D2 y más personajes de la saga de la Guerra de las Galaxias.",
        "title": "La guerra de las galaxias - Colección",
      },
      "english_name": "Spanish",
      "iso_3166_1": "ES",
      "iso_639_1": "es",
      "name": "Español",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Star Wars es una franquicia compuesta por una serie de películas concebidas primordialmentede por el cineasta estadounidense George Lucas, y producidas y distribuidas por The Walt Disney Company a partir de 2012. Su trama describe las vivencias de un grupo de personajes que habitan en una galaxia ficticia e interactúan con elementos como «la Fuerza», un campo de energía metafísico y omnipresente​ que posee un «lado oscuro» provocado por la ira, el miedo y el odio.",
        "title": "Star Wars - Colección",
      },
      "english_name": "Spanish",
      "iso_3166_1": "MX",
      "iso_639_1": "es",
      "name": "Español",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "مجموعه جنگ ستارگان",
      },
      "english_name": "Persian",
      "iso_3166_1": "IR",
      "iso_639_1": "fa",
      "name": "فارسی",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "Tähtien sota",
      },
      "english_name": "Finnish",
      "iso_3166_1": "FI",
      "iso_639_1": "fi",
      "name": "suomi",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Luke Skywalker, la Princesse Leia, Dark Vador, C3PO, R2D2 et de nombreux autres personnages du film sont désormais des noms familiers de l'un des projets cinématographiques les plus réussis de tous les temps.",
        "title": "Collection Star Wars",
      },
      "english_name": "French",
      "iso_3166_1": "CA",
      "iso_639_1": "fr",
      "name": "Français",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Luke Skywalker, la Princesse Leia, Dark Vador, C3PO, R2D2 et de nombreux autres personnages du film sont désormais des noms familiers de l'un des projets cinématographiques les plus réussis de tous les temps.",
        "title": "Star Wars - Saga",
      },
      "english_name": "French",
      "iso_3166_1": "FR",
      "iso_639_1": "fr",
      "name": "Français",
    },
    {
      "data": {
        "homepage": "",
        "overview": "מלחמת הכוכבים הוא שמה של סדרת סרטי מדע בדיוני מצליחים שיצר במאי הקולנוע ג'ורג' לוקאס. הסרט הראשון בסדרה, שנקרא אז \\"מלחמת הכוכבים\\" וכיום נקרא \\"מלחמת הכוכבים - פרק 4: תקווה חדשה\\", יצא בשנת 1977. מאז יצאו חמישה סרטים נוספים בסדרה, האחרון שבהם בשנת 2005. בסדרת הסרטים כיכבו אנסמבל של שחקנים כדוגמת מארק המיל, האריסון פורד, קארי פישר ואלק גינס (בטרילוגיה המקורית) והיידן כריסטנסן, נטלי פורטמן, יואן מקגרגור, ליאם ניסן, כריסטופר לי וסמואל ל. ג'קסון (בטרילוגיה המקדימה). בעקבות רכישת לוקאס פילם על ידי דיסני, הכריזו הרוכשים על הפקת טרילוגיה חדשה (פרקים 7-9); הראשון בהם צפוי לצאת לאקרנים בשנת 2015.",
        "title": "מלחמת הכוכבים - אוסף",
      },
      "english_name": "Hebrew",
      "iso_3166_1": "IL",
      "iso_639_1": "he",
      "name": "עִבְרִית",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "Ratovi zvijezda zbirka",
      },
      "english_name": "Croatian",
      "iso_3166_1": "HR",
      "iso_639_1": "hr",
      "name": "Hrvatski",
    },
    {
      "data": {
        "homepage": "",
        "overview": "A Csillagok háborúja (eredeti címe: Star Wars) egy űropera filmsorozatnak, irodalmi műveknek és számítógépes játékoknak a neve. Mindezek George Lucas filmrendező ötletei alapján készültek el. A filmek cselekménye a Galaktikus Köztársaság és a Galaktikus Birodalom közti átmenetek, valamint a Galaktikus Polgárháború (amely a Lázadó Szövetség és a Galaktikus Birodalom között zajlik) körül forog. Valójában az egész cselekmény a jó és a rossz harcát jelképezi. A történet „réges régen, egy messzi-messzi galaxisban…” játszódik. Az eredeti Csillagok háborúja-trilógia, főleg annak elsőként vetített része, az Egy új remény (IV. epizód) úttörő jelentőségű volt a sci-fi műfajában mind tartalmilag, mind a technikai megvalósítást tekintve, de az előzmény-trilógia (Baljós árnyak, A klónok támadása, A Sith-ek bosszúja) is számos újítást hozott magával.",
        "title": "Star Wars gyűjtemény",
      },
      "english_name": "Hungarian",
      "iso_3166_1": "HU",
      "iso_639_1": "hu",
      "name": "Magyar",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "",
      },
      "english_name": "Indonesian",
      "iso_3166_1": "ID",
      "iso_639_1": "id",
      "name": "Bahasa indonesia",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Guerre stellari (in inglese Star Wars) è un franchise creato da George Lucas, che si sviluppa da una saga cinematografica iniziata nel 1977 col film Guerre stellari, sottotitolato retroattivamente Episodio IV - Una nuova speranza. A questo film sono seguite altre due pellicole, distribuite a tre anni di distanza l'una dall'altra: L'Impero colpisce ancora (1980) e Il ritorno dello Jedi (1983). Questi tre film costituiscono la cosiddetta \\"trilogia originale\\". Sedici anni dopo l'uscita dell'ultimo film, Lucas decise di girare una trilogia prequel, composta da La minaccia fantasma (1999), L'attacco dei cloni (2002) e La vendetta dei Sith (2005). Nel 2012 The Walt Disney Company acquistò infine i diritti della serie e avviò la produzione di una trilogia sequel con Il risveglio della Forza (2015), Gli ultimi Jedi (2017) e L'ascesa di Skywalker (2019). Essa è stata alternata a una serie Anthology che include Rogue One: A Star Wars Story (2016) e Solo: A Star Wars Story (2018).",
        "title": "Guerre stellari - Collezione",
      },
      "english_name": "Italian",
      "iso_3166_1": "IT",
      "iso_639_1": "it",
      "name": "Italiano",
    },
    {
      "data": {
        "homepage": "",
        "overview": "\\"はるか昔、はるか彼方の銀河系を舞台に、さまざまな登場人物たちの冒険を描いた壮大なスペースオペラ劇場映画シリーズ....\\"",
        "title": "スター・ウォーズ シリーズ",
      },
      "english_name": "Japanese",
      "iso_3166_1": "JP",
      "iso_639_1": "ja",
      "name": "日本語",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "",
      },
      "english_name": "Georgian",
      "iso_3166_1": "GE",
      "iso_639_1": "ka",
      "name": "ქართული",
    },
    {
      "data": {
        "homepage": "",
        "overview": "오래전 멀고 먼 은하계에...",
        "title": "스타워즈 시리즈",
      },
      "english_name": "Korean",
      "iso_3166_1": "KR",
      "iso_639_1": "ko",
      "name": "한국어/조선말",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "Filmai „Žvaigždžių karai“",
      },
      "english_name": "Lithuanian",
      "iso_3166_1": "LT",
      "iso_639_1": "lt",
      "name": "Lietuvių",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "",
      },
      "english_name": "Norwegian Bokmål",
      "iso_3166_1": "NO",
      "iso_639_1": "nb",
      "name": "Bokmål",
    },
    {
      "data": {
        "homepage": "",
        "overview": "De collectie van de 9 theatrale productie van films uit de regisseur, scenarioschrijver en producer George Lucas. Luke Skywalker, Princess Leia, Darth Vader, C3PO, R2D2 en vele andere personages uit de film zijn nu vertrouwde namen van een van de meest succesvolle filmprojecten aller tijden.",
        "title": "Star Wars Collectie",
      },
      "english_name": "Dutch",
      "iso_3166_1": "NL",
      "iso_639_1": "nl",
      "name": "Nederlands",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "Star Wars - Den Komplette Samlingen",
      },
      "english_name": "Norwegian",
      "iso_3166_1": "NO",
      "iso_639_1": "no",
      "name": "Norsk",
    },
    {
      "data": {
        "homepage": "http://www.starwars.com/",
        "overview": "Seria filmów fantastycznych z gatunku space opera (science fantasy) George’a Lucasa.",
        "title": "Gwiezdne wojny - Kolekcja",
      },
      "english_name": "Polish",
      "iso_3166_1": "PL",
      "iso_639_1": "pl",
      "name": "Polski",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Uma série épica de filmes ópera espacial, que retratam as aventuras de vários personagens \\"há muito tempo em uma galáxia muito, muito distante ....\\"",
        "title": "Star Wars: Coleção",
      },
      "english_name": "Portuguese",
      "iso_3166_1": "BR",
      "iso_639_1": "pt",
      "name": "Português",
    },
    {
      "data": {
        "homepage": "https://www.starwars.com/",
        "overview": "Uma série épica de filmes ópera espacial, que retratam as aventuras de vários personagens \\"há muito tempo numa galáxia muito, muito distante...\\"",
        "title": "Star Wars - Guerra das Estrelas",
      },
      "english_name": "Portuguese",
      "iso_3166_1": "PT",
      "iso_639_1": "pt",
      "name": "Português",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "Colecția Star Wars",
      },
      "english_name": "Romanian",
      "iso_3166_1": "RO",
      "iso_639_1": "ro",
      "name": "Română",
    },
    {
      "data": {
        "homepage": "http://www.starwars.com",
        "overview": "«Звёздные войны» — эпическая фантастическая киносага, созданная Джорджем Лукасом. Эпизоды серии вобрали в себя архетипы как научной фантастики, так и классической мифологии, а также предоставили их великолепную музыкальную интерпретацию. Фильмы являются прекрасным примером космооперы, они заняли заметное место в поп-культуре и стали самой кассовой киносерией всех времён. В последние годы серия приобрела ещё большую популярность, породив новое поколение восторженных фанатов.",
        "title": "Звёздные войны (Коллекция)",
      },
      "english_name": "Russian",
      "iso_3166_1": "RU",
      "iso_639_1": "ru",
      "name": "Pусский",
    },
    {
      "data": {
        "homepage": "http://www.starwars.com",
        "overview": "",
        "title": "Star Wars (kolekcia)",
      },
      "english_name": "Slovak",
      "iso_3166_1": "SK",
      "iso_639_1": "sk",
      "name": "Slovenčina",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "колекција Ратови Звезда",
      },
      "english_name": "Serbian",
      "iso_3166_1": "RS",
      "iso_639_1": "sr",
      "name": "Srpski",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Star Wars är en saga som utspelar sig i en fjärran galax och där kampen mellan gott och ont står i centrum. I de tre första filmerna som producerades utkämpas ett inbördeskrig i galaxen där kampen står mellan det onda Rymdimperiet.",
        "title": "Star Wars (samling)",
      },
      "english_name": "Swedish",
      "iso_3166_1": "SE",
      "iso_639_1": "sv",
      "name": "svenska",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "สตาร์ วอร์ส คอลเลกชัน",
      },
      "english_name": "Thai",
      "iso_3166_1": "TH",
      "iso_639_1": "th",
      "name": "ภาษาไทย",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Yıldız Savaşları, George Lucas tarafından yapılmış, öncelikle filmleriyle tanınmış, sonraki yıllarda çizgi roman, bilgisayar ve konsol oyunları, televizyon yapımları vb. dallarda ününü geliştirmiş kurgusal evren ve markadır. Bugüne kadar 11 film bizlerle buluşsa da ana hikayeyi anlatan 8 film bulunmakta. İzleme sıralaması 4-5-6-1-2-3-7-8-9 şeklindedir.",
        "title": "Yıldız Savaşları [Seri]",
      },
      "english_name": "Turkish",
      "iso_3166_1": "TR",
      "iso_639_1": "tr",
      "name": "Türkçe",
    },
    {
      "data": {
        "homepage": "",
        "overview": "Культова епічна фантастична сага, що містить оповіді про протистояння планет, цивілізації та фракцій «Далекої галактики».",
        "title": "Зоряні війни | Колекція",
      },
      "english_name": "Ukrainian",
      "iso_3166_1": "UA",
      "iso_639_1": "uk",
      "name": "Український",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "Loạt phim Chiến Tranh Giữa Các Vì Sao",
      },
      "english_name": "Vietnamese",
      "iso_3166_1": "VN",
      "iso_639_1": "vi",
      "name": "Tiếng Việt",
    },
    {
      "data": {
        "homepage": "https://www.starwars.com/",
        "overview": "《星球大战》(英语：Star Wars)，中文可简称“星战”，是美国导演兼编剧乔治·卢卡斯所构思拍摄的一系列科幻电影。同时“星球大战”也是该系列电影最早拍摄上映的第四章故事的原始片名。从1970年代末到1980年代，卢卡斯出品了《星球大战》三部曲。他曾说明第一部星战（第四部曲）的人物及故事是参照日本导演黑泽明的《战国英豪》(隠し砦の三悪人)所创作而成；原三部曲以二战英德大战为架构并溶入美国西部片及日本武士刀剧的节奏；蕴涵了太空冒险、希腊神话的大视野制作。影片中所使用的特效技术重新定义并改变了往后太空科幻片的未来发展。1990年代末，基于各界要求又拍摄了原三部曲之前的故事，又称“星战前传”。同时他把最初的三部曲改为六部系列的第四、五、六集，并将最早的第四集改名为“新希望”（A New Hope）。2012年10月，华特迪士尼公司以40.5亿美元收购卢卡斯影业。该片描写，在很久以前，一个遥远的银河系，肩负正义使命的绝地武士与帝国邪恶黑暗势力作战的故事。本片开启好莱坞电影商品授权的庞大事业，旗下分别有小说、漫画、玩具与电玩游戏等相关周边产业，系列中每部电影，小说与电玩游戏，皆以“很久以前，在一个遥远的银河系……”(A long time ago in a galaxy far, far away...)开头。",
        "title": "星球大战（系列）",
      },
      "english_name": "Mandarin",
      "iso_3166_1": "CN",
      "iso_639_1": "zh",
      "name": "普通话",
    },
    {
      "data": {
        "homepage": "",
        "overview": "",
        "title": "《星球大戰》系列",
      },
      "english_name": "Mandarin",
      "iso_3166_1": "HK",
      "iso_639_1": "zh",
      "name": "普通话",
    },
    {
      "data": {
        "homepage": "",
        "overview": "在很久以前，一個遙遠的銀河系，肩負正義使命的絕地武士與帝國邪惡黑暗勢力作戰的故事。",
        "title": "星際大戰（系列）",
      },
      "english_name": "Mandarin",
      "iso_3166_1": "TW",
      "iso_639_1": "zh",
      "name": "普通话",
    },
  ],
}
`;
